### ISSUE TITLE
Accents

### ISSUE THREAD

**Issue by BrRoman** [bbloomf/jgabc#37]
Dear Benjamin,

I have performed a little "Accenteur", i.e. a JS application to accent latin texts ([https://github.com/BrRoman/accenteur](https://github.com/BrRoman/accenteur), test [here](https://brroman.github.io/accenteur/)).
It is quite stable, even if some errors still occur (Iésu, Mariǽ).

So I was wondering if you would be interested by this feature in your Transcription tool: the user, in mode "two boxes separated", would have a checkbox "Accents auto", so that when he types an inaccented word, your app would put the correct accent (or at least suggests accented forms).
I am not wed to this idea (in french: "Je n'y tiens pas mordicus"), and I have not a lot of time for that, but if your are interested, I would be glad to participate to the elaboration of this feature.

Yours in Christ
Brother Roman Mary

**Comment by bbloomf** [comment]
This is a good idea.  I'm not sure a checkbox would even be needed, since any word of 3+ syllable that doesn't have an accent marked may as well get marked.  Although, I suppose some accented syllables aren't marked because the vowel is a majuscule, and in these cases there'd be no way to override Accenteur's accenting if it were wrong (e.g., Anima could not be marked manually were Accenteur to accent it incorrectly, since we don't want to have to write Ánima)

Thank you for this suggestion.  I am not sure when I will have time to look at it, but it is definitely a good idea, and I will look into it when I get a chance and let you know if I need any help.

Thanks!

**Comment by pierrefrancois** [comment]
Hello

I am also happy with this feature.

I suppose that everybody who is using this tool will be interested to 
add the accents if not present, so I share the opinion of Benjamin a 
checkbox will not be needed.

I need to understand better how to override the automatic accents. When 
you see the word "tenere", I think you have two possibilities: "tenére" 
(infinitive of a verb) and "ténere" (adverb meaning dearly). So a manual 
intervention is necessary to correct the possible misbehaviour of Accenteur.

Does it mean that once an accent is set on a word, manually or not, the 
system does no longer try to put an accent on it? I don't understand the 
statement of Benjamin about "Anima" below: as long as the system doesn't 
change "Anima" (correct) in "Aníma" (incorrect) or whatever, everything 
goes fine. I hope I understood you correctly.

Kind regards.

Fr. Pierre

**Comment by bbloomf** [comment]
Yes, what I meant was that any word that had an accent marked manually would not be checked by accenteur.  (I suppose it could be nice to check it anyway, and if accenteur thought the accent should be on a different syllable, show the word in a different color.)  You would thus not be able to override a word that started with a capital vowel that has the accent, meaning that if accenteur were to incorrectly accent "Anima" as "Aníma" there would be no way to override it in this scheme except by writing "Ánima", which wouldn't be ideal.

Those ambiguous words like tenere could be highlighted and accented manually by tabbing through,  the way it currently works, each time you hit <TAB> while the "Text" text area has the focus, the next 3+ syllable unaccented word is selected, and you can press 1 or 2 to accent the antepenult or the penult of the word.

**Comment by BrRoman** [comment]
> (I suppose it could be nice to check it anyway, and if accenteur thought the accent should be on a different syllable, show the word in a different color.)

The checking of the "yet-accented-words" is an issue of Accenteur, and this discussion encourages me to fix it!

> Those ambiguous words like tenere could be highlighted and accented manually by tabbing through, **the way it currently works**, each time you hit while the "Text" text area has the focus, the next 3+ syllable unaccented word is selected, and you can press 1 or 2 to accent the antepenult or the penult of the word.

Nice feature! I hadn't seen it.
In the present state, accenteur opens a dialog and lets the user make his choice, but your system is easier.

Another sensible point to be considered: the core file (static/js/data.js) is very big (8 Mo) and will slow the page's loading.

> Thank you for this suggestion. I am not sure when I will have time to look at it, but it is definitely a good idea, and I will look into it when I get a chance and let you know if I need any help.

Thanks for your replies and enthusiasm, and I am at your disposal within the limits of my competences.

Yours in Christ
Brother Roman Mary
